tests/slurm_command_gen_strategy/test_sleep.py

```diff
@@ -0,0 +1,56 @@
+# SPDX-FileCopyrightText: NVIDIA CORPORATION & AFFILIATES
+# Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict, List
+
+import pytest
+from cloudai.schema.test_template.sleep.slurm_command_gen_strategy import SleepSlurmCommandGenStrategy
+from cloudai.systems import SlurmSystem
+
+
+class TestSleepSlurmCommandGenStrategy:
+    def get_test_command(
```

Comment:
I would suggest you reworking the pipeline:
1. `generate_test_command` is the function under test, it would be better to call it explicitly from the test function.
2. Command gen strategy can become a fixture.

```py
class TestSleepSlurmCommandGenStrategy:
    @pytest.fixture
    def cmd_gen_strategy(self, slurm_system: SlurmSystem) -> SleepSlurmCommandGenStrategy:
        return SleepSlurmCommandGenStrategy(slurm_system, {})

    def test_func(self, cmd_gen_strategy: SleepSlurmCommandGenStrategy):
        cmd_gen_strategy.generate_test_command(...)
```

_(this is the same for all new tests)_